components/core/src/clp/LibarchiveFileReader.cpp

```diff
@@ -252,6 +253,7 @@ ErrorCode LibarchiveFileReader::read_next_data_block() {
             &m_data_block_pos_in_file
     );
     if (ARCHIVE_OK != return_value) {
+        m_pos_in_data_block = m_data_block_length;
```

Comment:
Is there any guarantee what `m_data_block_length` would be set in a failure call? I didn't see any doc specifying this.
One safe way I would propose is that we set `m_data_block` to `nullptr`, and set both `m_data_block_length` and `m_data_block_pos_in_file` no matter which error happens.

Reply:
In this case it doesn't really matter if m_data_block_length is set.  setting m_pos_in_data_block = m_data_block_length simply indicates that there is no remaining data in the data block (since the pos points to the end of data block)

What I really care about here is the EOF case, where m_data_block_length should be set to 0. it would be more obvious to see what I intend to do if I set m_pos_in_data_block = 0. but I am not sure if there is any document saying that m_data_block_length is guranteed to be 0 when an EOF is returned.